### ISSUE TITLE
SearX API for JSON return empty "suggestions[]" and WEB does not.

### ISSUE THREAD

**Issue by Outstep** [searx/searx#1242]
Hi Again,

Just one more question about the SearX API and JSON return.

I did an test with:
curl "http://localhost:8888/?q=solar%20panels&categories=general&format=json&lang=en&pageno=1"| python -m json.tool

The JSON return has most of the information, but I noticed that the "suggestions : []" was blank

part of the JSON output shows:

        "positions": [
            10
        ],
        "pretty_url": "http://www.freesunpower.com/solarpanels.php",
        "score": 0.1,
        "title": "Basic Tutorials: Solar Panels for Solar \u2026",
        "url": "http://www.freesunpower.com/solarpanels.php"
    }
],
**"suggestions": [],**
"unresponsive_engines": [
    [
        "yacy",
        "request exception"
    ],
    [
        "gigablast",
        "unexpected crash: No JSON object could be decoded"
    ]
]

Actually, I do not know if image will come through, but on my we interface it does come through with some "SUGGESTIONS" in that section.

My question is as to why does the json "suggesions" not have the text for the suggestions that we web interface has since I would like to get that data as well, if possible.

Thanks again and it seems that I am almost there as far as having a metasearch engine that works really good.

Cheers,
Lonnie

**Comment by asciimoo** [comment]
@lonnieOST it doesn't return suggestions for me. The screenshot shows an infobox which is presented in the json too:
```
curl -q "http://localhost:8888/?q=solar%20panels&categories=general&format=json&lang=en&pageno=1" | python -m json.tool
[...]
    "infoboxes": [        
        {                                                
            "content": "Solar panels absorb sunlight as a source of energy to generate electricity or heat.",
            "engine": "wikipedia",                                                                                                                                           
            "id": "https://en.wikipedia.org/wiki/Solar_panel",
            "img_src": "https://upload.wikimedia.org/wikipedia/commons/thumb/b/b8/Photovoltaik_Dachanlage_Hannover_-_Schwarze_Heide_-_1_MW.jpg/300px-Photovoltaik_Dachanlage_Hannover_-_Schwarze_Heide_-_1_MW.jpg",
            "infobox": "Solar panel",                                                                                                                                                                                             
            "urls": [                                                                                               
                {                        
                    "title": "Wikipedia",                                                   
                    "url": "https://en.wikipedia.org/wiki/Solar_panel"                                                                                                        
                }            
            ]              
        }                         
    ]
[...]
```

**Comment by Outstep** [comment]
Hello,

That's the point. I would like to get back the suggestions in the JSON that
are coming up on the web display.

For me when I run the query with curl, I do not get any suggestions[]

-----------------------------------------

            "category": "general",
            "content": "Manufacturer of solar panels, cells, kits, and
travel solar electric products. We provide commercial solar sign & flood
lighting systems along with a large ...",
            "engine": "bing",
            "engines": [
                "bing"
            ],
            "parsed_url": [
                "http",
                "www.siliconsolar.com",
                "/",
                "",
                "",
                ""
            ],
            "positions": [
                10
            ],
            "pretty_url": "http://www.siliconsolar.com/",
            "score": 0.1,
            "title": "Solar Panels & Cells - Silicon Solar Store",
            "url": "http://www.siliconsolar.com/"
        }
    ],
   * "suggestions": [],*
    "unresponsive_engines": [
        [
            "yacy",
            "request exception"
        ],
        [
            "gigablast",
            "unexpected crash: No JSON object could be decoded"
        ]
    ]
}


-----------------------------------------

but when I do the query through the web interface then I do get the
suggestions.

How do I get those suggestions in my JSON query as well?

Thanks,
Lonnie

**Comment by asciimoo** [comment]
@lonnieOST I don't get any suggestions from the web ui (http://localhost:8888/?q=solar%20panels&categories=general) so the JSON response is correct.

**Comment by Outstep** [comment]
If you go to the web interface in a browser at http://localhost:8888 and
enter a query "solar panels" for example then with the Oscar theme you will
get a "suggestions" box on the right-hand side. That is what I mean as to
the web interface has suggestions and the JSON does not.

**Comment by asciimoo** [comment]
@lonnieOST there is no "suggestions" box for me:

![Uploading searsolarpanel.png…](https://i.imgur.com/Jpgfa3N.png)

**Comment by Outstep** [comment]
I see a suggestions box:

Lower right-hand side.
Cheers,
Lonnie

**Comment by Outstep** [comment]
This is also running on the latest SearX 0.14 , I think that it is.

**Comment by Pofilo** [comment]
Are you even with the 0.14 tag ?
Can you give us a screenshot ?

**Comment by Outstep** [comment]
Hello,

the change log that I have for this version shows:

0.14.0 2018.02.19
=================

- New theme: oscar-logicodev dark
- New engines

  - AskSteem (general)
- Autocompleter fix for py3
- Engine fixes (pdbe, pubmed, gigablast, google, yacy, bing videos,
microsoft academic)
- "All" option is removed from languages
- Minor UI changes
- Translation updates

**Comment by Outstep** [comment]
Hello All,

Not sure why my images are not showing up but the screenshot that I have does show a "Suggestions" section 

Let me try again to send over the screen shot. 0.14.0 2018.02.19

![screenshot from 2018-03-15 19-21-00](https://user-images.githubusercontent.com/24719646/37519938-7dbdfee4-28f1-11e8-9093-6c5f5820fb96.png)

**Comment by Outstep** [comment]
Why is the issue closed?

We seem to have one version showing suggestions and one version not showing suggestions for the same web search and in both cases the JSON call does not return any suggestions.

**Comment by Pofilo** [comment]
Okay I get it !
If you take a look at: [https://github.com/asciimoo/searx/search?utf8=%E2%9C%93&q=Suggestions&type=](https://github.com/asciimoo/searx/search?utf8=%E2%9C%93&q=Suggestions&type=), you can see that the _yahoo engine_ returns suggestions !

So if you have _yahoo_ activated, you'll get suggestions ! (just an example)

Going back to the problem of the issue, does the json research use at least one engine returning suggestions ?

**Comment by asciimoo** [comment]
@lonnieOST you have non-default engines enabled on your screenshot. Enable these engines when you create JSON request and it will return suggestions. The issue is closed, because searx works as expected. If you specify yahoo engine in the json request too, it returns suggestions: http://localhost:8888/?q=!yh%20solar%20panels&categories=general&format=json .

**Comment by Outstep** [comment]
Hello,

I see what you mean, had them off  as "default".  What I was doing was to
start up the default engines and then while the server was running and then
I would  enable "Yahoo" engine, for example and did not see JSON results
after that enabling in the preferences post-startup.

Thanks so very much as this really was making me pull my hair out trying to
determine what was happening.

Yes, the issue is closed and I sincerely appreciate all of the help.
Have a great weekend.
